### PR TITLE
feat(internal): add step.Skip() function

### PR DESCRIPTION
**Pull request by jbrockopp**
Part of the effort for https://github.com/go-vela/community/issues/54

This adds a new `step.Skip()` function to the `go-vela/pkg-executor/internal/step` package.

The intention of this function is to create the ruledata from the llbrary resources and verify the container for the step should be executed.

https://github.com/go-vela/pkg-executor/blob/621ca0a7bb301d3fd403d4a6e67820a97054a1a8/internal/step/skip.go#L15-L46

### CODE REVIEW

File: internal/step/skip_test.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+		ID:          "step_github_octocat_1_init",
+		Directory:   "/home/github/octocat",
+		Environment: map[string]string{"FOO": "bar"},
+		Image:       "#init",
+		Name:        "init",
+		Number:      1,
+		Pull:        "always",
+	}
+
+	_repo := &library.Repo{
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply internal/step/skip_test.go:108]: Ignoring this 👍

File: internal/step/skip_test.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+	"testing"
+
+	"github.com/go-vela/sdk-go/vela"
+	"github.com/go-vela/types/library"
+	"github.com/go-vela/types/pipeline"
+)
+
+func TestStep_Skip(t *testing.T) {
+	// setup types
+	_build := &library.Build{
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply internal/step/skip_test.go:17]: Ignoring this 👍

File: internal/step/skip_test.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+		Author:       vela.String("OctoKitty"),
+		Branch:       vela.String("master"),
+		Ref:          vela.String("refs/heads/master"),
+		BaseRef:      vela.String(""),
+		Host:         vela.String("example.company.com"),
+		Runtime:      vela.String("docker"),
+		Distribution: vela.String("linux"),
+	}
+
+	_deploy := &library.Build{
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply internal/step/skip_test.go:44]: Ignoring this 👍

File: internal/step/skip_test.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+		Author:       vela.String("OctoKitty"),
+		Branch:       vela.String("master"),
+		Ref:          vela.String("refs/heads/master"),
+		BaseRef:      vela.String(""),
+		Host:         vela.String("example.company.com"),
+		Runtime:      vela.String("docker"),
+		Distribution: vela.String("linux"),
+	}
+
+	_tag := &library.Build{
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply internal/step/skip_test.go:71]: Ignoring this 👍

**Review by wass3r — APPROVED** [review]

**Review by davidvader — APPROVED** [review]

**Pull request merged**